proxy/src/serverless/sql_over_http.rs

```diff
@@ -188,8 +188,14 @@ fn get_conn_info(
         .host_str()
         .ok_or(ConnInfoError::MissingHostname)?;
 
-    let endpoint =
-        endpoint_sni(hostname, &tls.common_names)?.ok_or(ConnInfoError::MalformedEndpoint)?;
+    let endpoint = if let Some(tls) = tls {
+        endpoint_sni(hostname, &tls.common_names)?.ok_or(ConnInfoError::MalformedEndpoint)?
+    } else {
+        hostname
+            .split_once(".")
```

Comment:
Will this always be name or could this end up being an IPv4 address?

Reply:
We won't support IP addresses in this field. It must always be a name. I'll make it return an error on IP addr.